### PR TITLE
Cleanup workflow code

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

## PR Context

Related #9570

Remove unneeded code related to workflows.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by PoshChan** [comment]
@iSazonov, your last commit had 4 failures in `PowerShell-CI-windows`
Workflow .Xaml module is not supported in PSCore.Import a XAML file directly should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1:323 char:11
    +         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1: line 323
323:         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML root module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1: line 327
327:         { Import-Module $xamlRootModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML nested module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1:331 char:11
    +         { Import-Module $xamlNestedModule -ErrorAction Stop } | Shoul ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1: line 331
331:         { Import-Module $xamlNestedModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Test-ModuleManifest tests.module manifest containing valid processed empty rootmodule file type fails: foo.xaml
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.TestModuleManifestCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidRootModuleInModuleManifest,Microsoft.PowerShell.Commands.TestModuleManifestCommand'. from D:\a\1\s\test\powershell\engine\Module\TestModuleManifest.Tests.ps1:92 char:11
    +         { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Module\TestModuleManifest.Tests.ps1: line 92
92:         { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop } | Should -Throw -ErrorId "$error,Microsoft.PowerShell.Commands.TestModuleManifestCommand"
```

**Comment by PoshChan** [comment]
@iSazonov, your last commit had 4 failures in `PowerShell-CI-linux`
Workflow .Xaml module is not supported in PSCore.Import a XAML file directly should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:323 char:11
    +         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 323
323:         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML root module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 327
327:         { Import-Module $xamlRootModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML nested module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:331 char:11
    +         { Import-Module $xamlNestedModule -ErrorAction Stop } | Shoul ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 331
331:         { Import-Module $xamlNestedModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Test-ModuleManifest tests.module manifest containing valid processed empty rootmodule file type fails: foo.xaml
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.TestModuleManifestCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidRootModuleInModuleManifest,Microsoft.PowerShell.Commands.TestModuleManifestCommand'. from /home/vsts/work/1/s/test/powershell/engine/Module/TestModuleManifest.Tests.ps1:92 char:11
    +         { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Module/TestModuleManifest.Tests.ps1: line 92
92:         { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop } | Should -Throw -ErrorId "$error,Microsoft.PowerShell.Commands.TestModuleManifestCommand"
```

**Comment by PoshChan** [comment]
@iSazonov, your last commit had 4 failures in `PowerShell-CI-macos`
Workflow .Xaml module is not supported in PSCore.Import a XAML file directly should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:323 char:11
    +         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 323
323:         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML root module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 327
327:         { Import-Module $xamlRootModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML nested module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:331 char:11
    +         { Import-Module $xamlNestedModule -ErrorAction Stop } | Shoul ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 331
331:         { Import-Module $xamlNestedModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Test-ModuleManifest tests.module manifest containing valid processed empty rootmodule file type fails: foo.xaml
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.TestModuleManifestCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidRootModuleInModuleManifest,Microsoft.PowerShell.Commands.TestModuleManifestCommand'. from /Users/vsts/agent/2.150.3/work/1/s/test/powershell/engine/Module/TestModuleManifest.Tests.ps1:92 char:11
    +         { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/engine/Module/TestModuleManifest.Tests.ps1: line 92
92:         { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop } | Should -Throw -ErrorId "$error,Microsoft.PowerShell.Commands.TestModuleManifestCommand"
```

**Comment by PoshChan** [comment]
@iSazonov, your last commit had 4 failures in `PowerShell-CI-linux`
Clean up open Runspaces when exit powershell process.PowerShell process should not freeze at exit
```powershell
Expected $true, but got $false.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Api/BasicEngine.Tests.ps1: line 53
53:         Wait-UntilTrue -sb { $process.HasExited } -TimeoutInMilliseconds 5000 -IntervalInMilliseconds 1000 | Should -BeTrue
```
Workflow .Xaml module is not supported in PSCore.Import a XAML file directly should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:323 char:11
    +         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 323
323:         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML root module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'System.NullReferenceException,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:327 char:11
    +         { Import-Module $xamlRootModule -ErrorAction Stop } | Should  ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 327
327:         { Import-Module $xamlRootModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML nested module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:331 char:11
    +         { Import-Module $xamlNestedModule -ErrorAction Stop } | Shoul ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 331
331:         { Import-Module $xamlNestedModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```

**Comment by PoshChan** [comment]
@iSazonov, your last commit had 3 failures in `PowerShell-CI-windows`
Workflow .Xaml module is not supported in PSCore.Import a XAML file directly should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1:323 char:11
    +         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1: line 323
323:         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML root module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'System.NullReferenceException,Microsoft.PowerShell.Commands.ImportModuleCommand'. from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1:327 char:11
    +         { Import-Module $xamlRootModule -ErrorAction Stop } | Should  ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1: line 327
327:         { Import-Module $xamlRootModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML nested module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1:331 char:11
    +         { Import-Module $xamlNestedModule -ErrorAction Stop } | Shoul ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Import-Module.Tests.ps1: line 331
331:         { Import-Module $xamlNestedModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```

**Comment by PoshChan** [comment]
@iSazonov, your last commit had 4 failures in `PowerShell-CI-macos`
Read-Host.[Console]::ReadKey() implementation on non-Windows.Should output correctly
```powershell
Expected 0, but got 1.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Host/Read-Host.Tests.ps1: line 25
25:             $LASTEXITCODE | Should -Be 0
```
Workflow .Xaml module is not supported in PSCore.Import a XAML file directly should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:323 char:11
    +         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 323
323:         { Import-Module $xamlFile -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML root module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'System.NullReferenceException,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:327 char:11
    +         { Import-Module $xamlRootModule -ErrorAction Stop } | Should  ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 327
327:         { Import-Module $xamlRootModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```
Workflow .Xaml module is not supported in PSCore.Import a module with XAML nested module should raise a 'NotSupported' error
```powershell
Expected an exception, with FullyQualifiedErrorId 'Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but the FullyQualifiedErrorId was 'Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand'. from /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1:331 char:11
    +         { Import-Module $xamlNestedModule -ErrorAction Stop } | Shoul ...
    +           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 331
331:         { Import-Module $xamlNestedModule -ErrorAction Stop } | Should -Throw -ErrorId "Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```

**Comment by iSazonov** [comment]
@SteveL-MSFT @PaulHigin Could you please review?

**Review by RDIL — APPROVED** [review]

File: src/System.Management.Automation/resources/Modules.resx [hunk 1]
```diff
@@ -141,15 +141,6 @@
   <data name="ManifestMemberNotFound" xml:space="preserve">
     <value>The module to process '{0}', listed in field '{1}' of module manifest '{2}' was not processed because no valid module was found in any module directory.</value>
   </data>
-  <data name="ScriptsFileNotFound" xml:space="preserve">
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/resources/Modules.resx]: Seems odd that some of this is removed when it's not workflow specific?

File: test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1 [hunk 2]
```diff
@@ -301,37 +301,6 @@ Describe "Import-Module should be case insensitive" -Tags 'CI' {
     }
 }
 
-Describe "Workflow .Xaml module is not supported in PSCore" -tags "Feature" {
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1]: Should we keep a single negative test case?  What error will get returned?

File: test/powershell/engine/Module/TestModuleManifest.Tests.ps1 [hunk 3]
```diff
@@ -81,8 +81,7 @@ Describe "Test-ModuleManifest tests" -tags "CI" {
     }
 
     It "module manifest containing valid processed empty rootmodule file type fails: <rootModuleValue>" -TestCases (
-        @{rootModuleValue = "foo.cdxml"; error = "System.Xml.XmlException"},  # fails when cmdlet tries to read it as XML
-        @{rootModuleValue = "foo.xaml"; error = "Modules_WorkflowModuleNotSupported"}   # not supported on PowerShell 6+
```
Comment (SteveL-MSFT) [inline test/powershell/engine/Module/TestModuleManifest.Tests.ps1]: Is there some other test case for invalid rootmodule files that xaml can be added to?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/resources/Modules.resx (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/resources/Modules.resx]: Yes, it is not related workflow but it is "dead" code so I clean up it too. It seems it make no sense to do this in another PR.

Comment (iSazonov) [reply src/System.Management.Automation/resources/Modules.resx]: Before the fix:
```
ipmo : Cannot load the workflow module 'XamlTest.xaml'. Workflow is not supported in PowerShell 6+.
At line:1 char:1
+ ipmo C:\temp\1\XamlRootModule.psd1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [Import-Module], NotSupportedException
+ FullyQualifiedErrorId : Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand
```

After the fix (there was a bug - fixed in new commit):
```
ipmo : The module to process 'XamlTest.xaml', listed in field 'NestedModules' of module manifest '' was not processed. The extension '.xaml' is not a valid module extension. The supported module extensions are '.dll', '.ps1', '.psm1', '.psd1' and '.cdxml'. Correct the extension then try adding the file 'XamlTest.xaml' again.
At line:1 char:1
+ ipmo C:\temp\1\XamlRootModule.psd1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [Import-Module], PSInvalidOperationException
+ FullyQualifiedErrorId : Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand
```

File: test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1 (hunk 2 above)
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1]: Before the fix:
```
ipmo : Cannot load the workflow module 'XamlTest.xaml'. Workflow is not supported in PowerShell 6+.
At line:1 char:1
+ ipmo C:\temp\1\XamlRootModule.psd1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [Import-Module], NotSupportedException
+ FullyQualifiedErrorId : Modules_WorkflowModuleNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand
```

After the fix (there was a bug - fixed in new commit):
```
ipmo : The module to process 'XamlTest.xaml', listed in field 'NestedModules' of module manifest '' was not processed. The extension '.xaml' is not a valid module extension. The supported module extensions are '.dll', '.ps1', '.psm1', '.psd1' and '.cdxml'. Correct the extension then try adding the file 'XamlTest.xaml' again.
At line:1 char:1
+ ipmo C:\temp\1\XamlRootModule.psd1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [Import-Module], PSInvalidOperationException
+ FullyQualifiedErrorId : Modules_InvalidModuleExtension,Microsoft.PowerShell.Commands.ImportModuleCommand
```

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1]: It seems we do the same previously with conclusion that make no sense to create special information exception for every feature which we don't support now.

File: test/powershell/engine/Module/TestModuleManifest.Tests.ps1 (hunk 3 above)
Comment (iSazonov) [reply test/powershell/engine/Module/TestModuleManifest.Tests.ps1]: Sorry, I don't understand your request. Do you want to add some new tests? If we have gaps in the test we could address this later.

**Comment by iSazonov** [comment]
Rebased to move to .Net Core 3.0 and get latest CIs.

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Pull request merged**